spec/Dynamic-Validator.md

```diff
@@ -0,0 +1,156 @@
+
+### Minimum Deposit
+TBD
+
+## Delegation
+The stakeholders have the right to choose validators as much as their shares.
+It's called a delegation.
+And the stakeholders who have been delegated called delegators.
+The delegation is valid only when the delegatee is not an eligible or not banned.
+The delegated stakes are returned when the account becomes an eligible or banned account.
+
+## Election
+The process that elects validators of a term is called the election.
+The election selects validators as the following rule.
+
+1. TBD
+
+## Voting Power
+Each elected validators has different voting power.
+The voting power is based on the delegation that the validator has received at the election.
+The block is valid only if the sum of voting power is more than 2/3 of the total delegations that elected validators received.
+
+## Validator Reward
+The block proposer gets the express fee of the blocks at the end of a term.
+Validators can get the reward after **WITHDRAW_DELAY** terms.
+However, the proposers cannot get all reward if he is not loyal to their duty.
+The reward is decreased according to the rate of the blocks the validator misses to sign.
+TBD: The rate of decreasing.
+
+## Punishment for Validators
+### Downtime
+The validator who doesn't produce blocks is jailed for a while.
+The jailed account cannot be a candidate during **CUSTODY_PERIOD**.
+*SELF_NOMINATE* transaction of the account is rejected.
+Actually, this is not a punishment.
+It is to give validators time to fix the nodes they manage.
+The jailed account can nominate itself again after **CUSTODY_PERIOD**.
+
+### Disloyal Validators
+CodeChain gives a penalty to the validator who doesn't participate in signing the blocks proposed by another node.
+See [Validator Reward](#Validator-Reward) for more information.
+
+### Double Vote
+CodeChain bans the account who double voted.
+The deposit and the reward the criminal earns is slashed and is given to the informant reporting the double vote.
+
+## Transactions
+### SELF_NOMIATION
+* quantity
+* metadata(TBD)
+
+This transaction registers the sender to candidate when the sum of the deposit is larger than **MIN_DEPOSIT**.
+The nomination is valid in **NOMINATE_EXPIRATION**.
+
+The account cannot withdraw the deposit manually.
+It's returned automatically when the account becomes an eligible account.
+
+### WITHDRAW
+* quantity
+
+This transaction withdraws the reward that the node earns as a validator.
+But the validator cannot withdraw the reward before **WITHDRAW_DELAY** passes.
+
+The transaction that tries to withdraw more than the account can fails.
+
+### DELEGATION
+* delegatee
+* quantity
+
+It's a transaction used by the stakeholders to select the validators.
+The validator can delegate the stake as much as it has stakes.
+The validator can delegate to any candidates, validators and jailed accounts.
+The delegations return automatically when the delegatee becomes eligible or banned.
```

Comment:
It's a transaction used by the stakeholders to select the validators.
The validator can delegate as much stakes as they have.
The validator can delegate any candidates, including validators and jailed accounts.
The delegations return automatically when the delegatee becomes eligible or banned.